driver/labjack/reader.h

```diff
@@ -0,0 +1,367 @@
+        double range = 10.0;
+        int neg_chan = 199;
+        int pos_chan;
+        ///@brief channel type (e.g. AIN, DIN, TC)
+        std::string channel_type = "";
+        ///@brief Thermocouple configuration if applicable
+        TCConfig tc_config;
+        ReaderChannelConfig() = default;
+
+        explicit ReaderChannelConfig(config::Parser &parser)
+                :   enabled(parser.optional<bool>("enabled", true)),
+                    data_type(parser.optional<std::string>("data_type", "float32")),
+                    key(parser.required<uint32_t>("channel")),
+                    range(parser.optional<double>("range", 10.0)),
+                    neg_chan(parser.optional<double>("neg_chan", SINGLE_ENDED)),
+                    pos_chan(parser.optional<int>("pos_chan",0)),
+                    channel_type(parser.optional<std::string>("type", "")),
+                    location(parser.optional<std::string>("port", "")){
+            if(!parser.ok())
+                LOG(ERROR) << "Failed to parse reader channel config: " << parser.error_json().dump(4);
+
+            if(this->channel_type == "TC") {
+                this->tc_config = TCConfig(parser);
+                // temparature : AIN#_EF_READ_A register
+                // voltage     : AIN#_EF_READ_B register
+                // CJC temp    : AIN#_EF_READ_C register
+                this->location = "AIN" + std::to_string(this->tc_config.pos_chan) + "_EF_READ_A";
+            }
+        }
+    };
+
+
+    ///////////////////////////////////////////////////////////////////////////////////
+    //                                   ReaderConfig                                //
+    ///////////////////////////////////////////////////////////////////////////////////
+    struct ReaderConfig {
+        ///@brief The type of device (e.g. T4, T7, T8, etc.)
+        std::string device_type;
+        ///@brief Key of device on synnax server
+        std::string device_key;
+        std::vector<ReaderChannelConfig> channels;
+        synnax::Rate sample_rate = synnax::Rate(1);
+        synnax::Rate stream_rate = synnax::Rate(1);
+        synnax::ChannelKey task_key;
+        std::set<uint32_t> index_keys;
+        std::string serial_number; // used to open devices
+        ///@brief The type of connection (i.e. USB, Ethernet, or WIFI)
+        std::string connection_type; // used to open devices
+        ///@brief map of locations on device to synnax channel keys
+        std::map<std::string, uint32_t> channel_map;
+        std::vector<std::string> phys_channels;
+        std::vector<ReaderChannelConfig> tc_channels;
+        std::set<uint32_t> tc_index_keys;
+        ///@brief whether to persist data to disk
+        bool data_saving;
+
+        ReaderConfig() = default;
+
+        explicit ReaderConfig(config::Parser &parser)
+                : device_type(parser.optional<std::string>("type", "")),
+                  device_key(parser.required<std::string>("device")),
```

Comment:
Any reason we use 'device_key' here instead of just device? I think we use device for all the other tasks